core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1731,6 +1764,57 @@ object Parser {
     */
   def end: Parser0[Unit] = Impl.EndParser
 
+  /** return the input string we are parsing.
+    */
+  def input: Parser0[String] = Impl.Input
```

Comment:
I'm a bit nervous about exposing this.

It closes off the possibility of ever easily supporting Streaming.

You could imagine parsing `Stream[String]` or `LazyList[String]`, but if you have people reaching for the whole String, that is going to blow up their memory.

On the other hand, I guess a streaming parser is just really different from a parser that parses a whole file. A streaming parser should be something like: `Stream[F, String] => Stream[ParserF[F, *], A]` or something...

Reply:
I see. In Megaparsec they used a lazily updated state field that stores source position and a snapshot of the stream.